server/util/capabilities/capabilities.go

```diff
@@ -55,7 +53,7 @@ func IsGranted(ctx context.Context, env environment.Env, cap akpb.ApiKey_Capabil
 	}
 	user, err := a.AuthenticatedUser(ctx)
 	if err != nil {
-		if isAnonymousUser := gstatus.Code(err) == codes.PermissionDenied; isAnonymousUser {
+		if isAnonymousUser := status.IsPermissionDeniedError(err) || status.IsUnauthenticatedError(err); isAnonymousUser {
```

Comment:
@bduffany probably knows better here -- but do we need both or would it be better if isAnonymous were only true for unauthenticated errors?

Reply:
I think we should probably go through at some point and audit all uses for PermissionDenied vs Unauthenticated and make sure they conform to the status proto comment's definitions. This PR was just focused on the GetUser/CreateUser flows.

I think once we're confident that we're returning the right statuses in the right places, we can probably switch this (and the two other places where we do these kinds of checks) to just check for Unauthenticated. Until we do that, I think changing this could have unintended consequences.

Reply:
I agree with not doing it in this PR -- and I also agree that we should make the anonymous user case a bit more clean and less error-prone eventually. Maybe we could treat anonymous users as just a special kind of user with the `ANON` (or empty) group ID and then the anonymous concept is no longer needed (except in some rarer cases where we want to prevent anonymous users from doing certain things... in those cases we could check that the group ID is `ANON`). I think @vadimberezniker also was looking at some related logic semi-recently and he might have thoughts as well.

Reply:
Yeah, we can have this discussion separately from this PR - I'd like to keep the scope small.